c2rust-analyze/src/borrowck/mod.rs

```diff
@@ -204,10 +205,12 @@ fn run_polonius<'tcx>(
     let mut facts = AllFacts::default();
     let mut maps = AtomMaps::default();
 
-    // Start the origin counter at 3.  This has no effect on the semantics, but makes for easier
-    // diffs between our facts and the facts generated by rustc.
-    for _ in 0..3 {
-        let _ = maps.origin();
+    // the 'static region is always 0
+    let static_origin = maps.origin();
+    // polonius gives an origin to each argument
+    let mut arg_origins = vec![];
+    for _ in 0..mir.arg_count {
+        arg_origins.push(maps.origin());
```

Comment:
Rustc assigns an origin to each lifetime parameter of the function, not to each value parameter, which is what `Body::arg_count` counts.

Reply:
Makes sense, I totally misread your [comment](https://github.com/immunant/c2rust/pull/975#issuecomment-1629506921) explaining this.

Reply:
I've made a change in dd458e0e